osquery/database/tests/database_tests.cpp

```diff
@@ -149,18 +157,23 @@ TEST_F(DatabaseTests, test_ptree_to_rj_object) {
   pt::ptree pt_in;
   pt_in.put("k", "v");
 
-  JSON rj_in = JSON::newObject();
-  rj_in.add("k", "v");
+  JSON rj_in = JSON::newArray();
+  JSON rj_obj = rj_in.newObject();
+  rj_obj.add("k", "v");
+  rj_in.push(rj_obj.doc(), rj_in.doc());
 
   ptree_to_json(pt_in, rj_in);
 }
 
 TEST_F(DatabaseTests, test_ptree_to_rj_array) {
   /* '{ "" : "a", "" : "b" }' */
 
-  pt::ptree pt_in;
-  pt_in.put("", "a");
-  pt_in.put("", "b");
+  pt::ptree pt_in, child1, child2, children;
+  child1.put("", "a");
+  child2.put("", "b");
+
+  pt_in.add_child("", child1);
+  pt_in.add_child("", child2);
```

Comment:
Isn't this going to create a `{ "" : { "" : "a" } , "" : { "" : "b" } }` which should be converted to `[ ["a"] , ["b"] ]`?

Reply:
Same as above.